spec/integ/sliding-sync-sdk.spec.ts

```diff
@@ -928,4 +928,95 @@ describe("SlidingSyncSdk", () => {
             expect(room.getMember(selfUserId)?.typing).toEqual(false);
         });
     });
+
+    describe("ExtensionReceipts", () => {
+        let ext: Extension;
+
+        beforeAll(async () => {
+            await setupClient();
+            const hasSynced = sdk!.sync();
+            await httpBackend!.flushAllExpected();
+            await hasSynced;
+            ext = findExtension("receipts");
+        });
+
+        it("gets enabled on the initial request only", () => {
+            expect(ext.onRequest(true)).toEqual({
+                enabled: true,
+            });
+            expect(ext.onRequest(false)).toEqual(undefined);
+        });
+
+        it("processes receipts", async () => {
+            const roomId = "!room:id";
+            const alice = "@alice:alice";
+            const lastEvent = mkOwnEvent(EventType.RoomMessage, { body: "hello" });
+            mockSlidingSync!.emit(SlidingSyncEvent.RoomData, roomId, {
+                name: "Room with receipts",
+                required_state: [],
+                timeline: [
+                    mkOwnStateEvent(EventType.RoomCreate, { creator: selfUserId }, ""),
+                    mkOwnStateEvent(EventType.RoomMember, { membership: "join" }, selfUserId),
+                    mkOwnStateEvent(EventType.RoomPowerLevels, { users: { [selfUserId]: 100 } }, ""),
+                    {
+                        type: EventType.RoomMember,
+                        state_key: alice,
+                        content: { membership: "join" },
+                        sender: alice,
+                        origin_server_ts: Date.now(),
+                        event_id: "$alice",
+                    },
+                    lastEvent,
+                ],
+                initial: true,
+            });
+            const room = client!.getRoom(roomId)!;
+            expect(room).toBeDefined();
+            expect(room.getReadReceiptForUserId(alice, true)).toBeNull();
+            ext.onResponse({
+                rooms: {
+                    [roomId]: {
+                        type: EventType.Receipt,
+                        content: {
+                            [lastEvent.event_id]: {
+                                "m.read": {
+                                    [alice]: {
+                                        ts: 1234567,
+                                    },
+                                },
+                            },
+                        },
+                    },
+                },
+            });
```

Comment:
The object that's used here is reused on line 1003. It looks to me like tests in this area might quite often have to generate this object, so I was wondering if you think it warrants making a tiny util function that takes in the `roomId,  eventId, username` and returns the required object?

Reply:
I don't anticipate using this object a bunch, but given sonarcloud is whining I probably will be forced to make a util function for it.